hwy/ops/generic_ops-inl.h

```diff
@@ -1249,6 +1250,43 @@ HWY_API V PopulationCount(V v) {
 }
 #endif
 
+HWY_API Vec128<int64_t,1> operator*(Vec128<int64_t,1> x, Vec128<int64_t,1> y) {
+    return Set(Simd<int64_t,1,0>(), GetLane(x)*GetLane(y));
+}
+HWY_API Vec128<uint64_t,1> operator*(Vec128<uint64_t,1> x, Vec128<uint64_t,1> y) {
+    return Set(Simd<uint64_t,1,0>(), GetLane(x)*GetLane(y));
+}
+
+#ifndef HWY_NATIVE_I64MULLO
+
+
+HWY_API Vec128<uint64_t> operator*(Vec128<uint64_t> x, Vec128<uint64_t> y) {
+  Vec128<uint32_t> x32 = BitCast(Simd<uint32_t,4,0>(), x);
+  Vec128<uint32_t> y32 = BitCast(Simd<uint32_t,4,0>(), y);
+  Vec128<uint64_t> lolo = MulEven(x32, y32);
+  Vec128<uint32_t> lohi = BitCast(Simd<uint32_t,4,0>(), MulEven(x32, BitCast(Simd<uint32_t,4,0>(), ShiftRight<32>(y))));
+  Vec128<uint32_t> hilo = BitCast(Simd<uint32_t,4,0>(), MulEven(BitCast(Simd<uint32_t,4,0>(), ShiftRight<32>(x)), y32));
+  return lolo + ShiftLeft<32>(BitCast(Simd<uint64_t,2,0>(), lohi + hilo));
+}
+HWY_API Vec128<int64_t> operator*(Vec128<int64_t> x, Vec128<int64_t> y) {
+  return BitCast(Simd<int64_t,2,0>(), BitCast(Simd<uint64_t,2,0>(), x) * BitCast(Simd<uint64_t,2,0>(), y));
+}
+#if HWY_TARGET == HWY_AVX2
+
+HWY_API Vec256<uint64_t> operator*(Vec256<uint64_t> x, Vec256<uint64_t> y) {
```

Comment:
I think this can share the same implementation as the Vec128 version if we make it a template with `<class V>` argument, where V is the arg and return type.
To get the `Simd<>` you can then use `DFromV<V> d32` and `RepartitionToWide<decltype(d32)> d64`.

(This assumes the Vec512 version is not a template, which it currently isn't. If it were, it might be ambiguous.)

Reply:
How do I restrict it to only be for 64 bit integers?
```c++
template<class V>
HWY_API V operator*(V x, V y){
    ...
}
```
How do I restrict this to be called for `V` such that `LaneType<V>` is an 8 byte integer?

Reply:
Ah yes, I should have mentioned that :)
`template <class V, HWY_IF_LANE_SIZE_V(V, 8)>` will do that.

Reply:
I looked through the source, and tried to follow the approach you took elsewhere.